### DISCUSSION TITLE
Project 4 Diversity Functions

### DISCUSSION THREAD

**Discussion by mhuzzell** [IBAHCM/RPiR#560] [Q&A for project] [answered]
I was a little confused by the instructions for Practical 4, so just wanted to clarify. The text says:

"First, you probably want to copy your existing **functions** again. You will edit **these functions** (renaming **them** as you go along) to create final, definitive diversity **functions** that do everything we want.

**Final diversity functions for a single value of q**
Add a test in the general diversity function that checks that there is only one value of q
and stop()s if there is more than one – now we have identified a problem in our code, we want to avoid it happening again! We will stop worrying about the specific (e.g. species richness) functions **now**, as it’s boring to keep on rewriting them!"

(Emphasis mine, apart from the part that's a header.)

There's only one general diversity function to copy over from Project 3 and edit for Project 4, plus one completely new function for Project 4. The demo only uses these two functions. Parsing through the instructions sequentially, though, it sounds like we are supposed to be creating p4_species_richness, p4_simpson_index, etc. -- just running through the newly edited and final general_diversity function instead of p3_general_diversity -- but then just leave them in the R folder for posterity, and not use them in the demo? Or should we understand this to mean that we should just create the new general_diversity function and not worry about creating new versions of the four specific diversity functions?

**Comment by richardreeve** [comment, marked as answer]
> Parsing through the instructions sequentially, though, it sounds like we are supposed to be creating p4_species_richness, p4_simpson_index, etc. -- just running through the newly edited and final general_diversity function instead of p3_general_diversity -- but then just leave them in the R folder for posterity, and not use them in the demo?

Yes, that's right.
